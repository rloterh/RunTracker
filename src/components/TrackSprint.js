import React, { useState } from 'react';
import axios from 'axios';
import { useDispatch } from 'react-redux';
import { getSprint } from '../actions';

const TrackSprint = () => {
  const [targetValue, setTarget] = useState(100);
  const [hoursValue, setHours] = useState(0);
  const [minutesValue, setMinutes] = useState(0);
  const [secondsValue, setSeconds] = useState(0);
  const [distanceValue, setDistance] = useState(0);
  const [remarksValue, setRemarks] = useState('');
  // const [effortValue, setEffort] = useState(0);

  const dispatch = useDispatch();

  const onSubmit = e => {
    e.preventDefault();
    const effortValue = ((distanceValue / targetValue) * 100);

    axios.post('http://localhost:3000/api/v1/sprints', {
      target: targetValue,
      hours: hoursValue,
      minutes: minutesValue,
      seconds: secondsValue,
      distance: distanceValue,
      remarks: remarksValue,
      effort: effortValue,
    });
    dispatch(getSprint());
  };

  return (
    <div>
      <div>

        <form onSubmit={onSubmit}>
          <select
            ref={targetInput}
            size="2"
            type="number"
            name="target"
            value={targetValue}
            onChange={e => setTarget(e.target.value)}
          >
          </select>
          <input
            ref={hoursInput}
            type="number"
            name="hours"
            min="0"
            max="24"
            placeholder="hours"
            className="hidden"
            value={hoursValue}
            onChange={e => setHours(e.target.value)}
          />
          <input
            ref={minsInput}
            type="number"
            name="minutes"
            min="0"
            max="60"
            placeholder="minutes"
            value={minutesValue}
            className="hidden"
            onChange={e => setMinutes(e.target.value)}
          />
          <input
            ref={secondsInput}
            type="number"
            name="seconds"
            min="0"
            max="60"
            placeholder="seconds"
            className="hidden"
            value={secondsValue}
            onChange={e => setSeconds(e.target.value)}
          />
          <input
            ref={distanceInput}
            type="number"
            name="distance"
            min="10"
            max="10000"
            placeholder="distance"
            className="hidden"
            value={distanceValue}
            onChange={e => setDistance(e.target.value)}
          />
          <textarea
            ref={remarksInput}
            name="remarks"
            placeholder="remarks"
            min="2"
            max="100"
            className="hidden"
            cols="20"
            rows="6"
            value={remarksValue}
            required
            onChange={e => setRemarks(e.target.value)}
          />
          <input
            ref={effortInput}
            type="hidden"
            name="effort"
            className="hidden"
            // value={effortValue}
          />

          <div ref={firstButtons}>
            <button
              type="button"
              onClick={() => goToSecondPage()}
            >
              Next
            </button>
          </div>

          <div ref={secondButtons} className="hidden">
            <button
              type="button"
              onClick={() => goToFirstPage()}
            >
              Prev
            </button>
            <button
              type="button"
              onClick={() => goToThirdPage()}
            >
              Next
            </button>
          </div>
 
          <div ref={thirdButtons} className="hidden">
            <button
              type="button"
              onClick={() => goToSecondPage()}
            >
              Prev
            </button>
            <button
              type="button"
              onClick={() => goToFourthPage()}
            >
              Next
            </button>
          </div>
 
          <div ref={fourthButtons} className="hidden">
            <button
              type="button"
              onClick={() => goToThirdPage()}
            >
              Prev
            </button>
            <button
              type="button"
              onClick={() => goToFifthPage()}
            >
              Next
            </button>
          </div>

          <div ref={fifthButtons} className="hidden">
            <button
              type="button"
              onClick={() => goToFourthPage()}
            >
              Prev
            </button>
            <button
              type="button"
              onClick={() => goToSixthPage()}
            >
              Next
            </button>
          </div>

          <div ref={sixthButtons} className="hidden">
            <button
              type="button"
              onClick={() => goToFifthPage()}
            >
              Prev
            </button>
            <button
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default TrackSprint;
