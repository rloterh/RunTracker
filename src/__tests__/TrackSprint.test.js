import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import TrackSprint from '../components/TrackSprint';
import store from '../redux/configureStore';

describe('checks if TrackSprint component renders correctly', () => {
  test('should renders correct text', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <TrackSprint />
        </BrowserRouter>
      </Provider>,
    );
    const text = screen.getByText('What\'s your sprint target?');
    expect(text).toBeInTheDocument();
  });

  test('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <TrackSprint />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
