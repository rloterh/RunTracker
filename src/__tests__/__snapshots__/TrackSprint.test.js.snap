// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if TrackSprint component renders correctly should render correctly 1`] = `
<div
  className="text-center h-full"
>
  <div
    className="p-4 bg-sblue text-white font-sans w-full"
  >
    <h1
      className="text-2xl font-black"
    >
      Record Sprint
    </h1>
  </div>
  <div
    className="p-4 border-b-2"
  >
    <div>
      <h3
        className="font-bold text-gray-600 font-sans"
      >
        What's your sprint target?
      </h3>
    </div>
  </div>
  <div
    className="flex flex-col items-center justify-center space-y-4"
  >
    <div
      className="flex flex-col items-center justify-center py-10 font-sans"
    >
      <section
        className="hidden w-3/4"
      >
        <svg
          className="CircularProgressbar "
          data-test-id="CircularProgressbar"
          style={Object {}}
          viewBox="0 0 100 100"
        >
          <path
            className="CircularProgressbar-trail"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fillOpacity={0}
            strokeWidth={8}
            style={
              Object {
                "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                "strokeDashoffset": "0px",
              }
            }
          />
          <path
            className="CircularProgressbar-path"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fillOpacity={0}
            strokeWidth={8}
            style={
              Object {
                "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                "strokeDashoffset": "289.02652413026095px",
              }
            }
          />
          <text
            className="CircularProgressbar-text"
            style={Object {}}
            x={50}
            y={50}
          >
            0 h
          </text>
        </svg>
      </section>
      <section
        className="hidden w-3/4"
      >
        <svg
          className="CircularProgressbar "
          data-test-id="CircularProgressbar"
          style={Object {}}
          viewBox="0 0 100 100"
        >
          <path
            className="CircularProgressbar-trail"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fillOpacity={0}
            strokeWidth={8}
            style={
              Object {
                "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                "strokeDashoffset": "0px",
              }
            }
          />
          <path
            className="CircularProgressbar-path"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fillOpacity={0}
            strokeWidth={8}
            style={
              Object {
                "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                "strokeDashoffset": "289.02652413026095px",
              }
            }
          />
          <text
            className="CircularProgressbar-text"
            style={Object {}}
            x={50}
            y={50}
          >
            0 mins
          </text>
        </svg>
      </section>
      <section
        className="hidden w-3/4"
      >
        <svg
          className="CircularProgressbar "
          data-test-id="CircularProgressbar"
          style={Object {}}
          viewBox="0 0 100 100"
        >
          <path
            className="CircularProgressbar-trail"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fillOpacity={0}
            strokeWidth={8}
            style={
              Object {
                "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                "strokeDashoffset": "0px",
              }
            }
          />
          <path
            className="CircularProgressbar-path"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fillOpacity={0}
            strokeWidth={8}
            style={
              Object {
                "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                "strokeDashoffset": "289.02652413026095px",
              }
            }
          />
          <text
            className="CircularProgressbar-text"
            style={Object {}}
            x={50}
            y={50}
          >
            0 s
          </text>
        </svg>
      </section>
      <section
        className="hidden w-3/4"
      >
        <svg
          className="CircularProgressbar "
          data-test-id="CircularProgressbar"
          style={Object {}}
          viewBox="0 0 100 100"
        >
          <path
            className="CircularProgressbar-trail"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fillOpacity={0}
            strokeWidth={8}
            style={
              Object {
                "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                "strokeDashoffset": "0px",
              }
            }
          />
          <path
            className="CircularProgressbar-path"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fillOpacity={0}
            strokeWidth={8}
            style={
              Object {
                "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                "strokeDashoffset": "289.02652413026095px",
              }
            }
          />
          <text
            className="CircularProgressbar-text"
            style={Object {}}
            x={50}
            y={50}
          >
            0 m
          </text>
        </svg>
      </section>
    </div>
    <form
      className="space-y-10 absolute bottom-20 font-sans"
      onSubmit={[Function]}
    >
      <select
        className="focus:outline-none w-full"
        name="target"
        onChange={[Function]}
        size="2"
        type="number"
        value={100}
      >
        <option
          value={100}
        >
          100
           
          metres
           
        </option>
        <option
          value={200}
        >
          200
           
          metres
           
        </option>
        <option
          value={500}
        >
          500
           
          metres
           
        </option>
        <option
          value={1000}
        >
          1000
           
          metres
           
        </option>
        <option
          value={2000}
        >
          2000
           
          metres
           
        </option>
        <option
          value={5000}
        >
          5000
           
          metres
           
        </option>
      </select>
      <input
        className="hidden"
        max="24"
        min="0"
        name="hours"
        onChange={[Function]}
        type="number"
        value={0}
      />
      <input
        className="hidden"
        max="60"
        min="0"
        name="minutes"
        onChange={[Function]}
        placeholder="minutes"
        type="number"
        value={0}
      />
      <input
        className="hidden"
        max="60"
        min="0"
        name="seconds"
        onChange={[Function]}
        placeholder="seconds"
        type="number"
        value={0}
      />
      <input
        className="hidden"
        max="10000"
        min="10"
        name="distance"
        onChange={[Function]}
        placeholder="distance"
        type="number"
        value={0}
      />
      <textarea
        className="hidden"
        cols="20"
        max="100"
        min="2"
        name="remarks"
        onChange={[Function]}
        placeholder="remarks"
        required={true}
        rows="6"
        value=""
      />
      <input
        className="hidden"
        name="effort"
        type="hidden"
      />
      <div
        className="bg-gray-100 w-full"
      >
        <button
          className="px-10 py-3 bg-sgreen text-white w-full"
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
      <div
        className="hidden"
      >
        <button
          className="px-10 py-3 text-gray-600 w-full"
          onClick={[Function]}
          type="button"
        >
          Prev
        </button>
        <button
          className="px-10 py-3 bg-sgreen text-white w-full"
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
      <div
        className="hidden"
      >
        <button
          className="px-10 py-3 text-grey-600 w-full text-gray-600"
          onClick={[Function]}
          type="button"
        >
          Prev
        </button>
        <button
          className="px-10 py-3 bg-sgreen text-white w-full"
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
      <div
        className="hidden"
      >
        <button
          className="px-10 py-3 text-gray-600 w-full"
          onClick={[Function]}
          type="button"
        >
          Prev
        </button>
        <button
          className="px-10 py-3 bg-sgreen text-white w-full"
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
      <div
        className="hidden"
      >
        <button
          className="px-10 py-3 text-gray-600 w-full"
          onClick={[Function]}
          type="button"
        >
          Prev
        </button>
        <button
          className="px-10 py-3 bg-sgreen text-white w-full"
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
      <div
        className="hidden"
      >
        <button
          className="px-10 py-3 text-gray-600 w-full"
          onClick={[Function]}
          type="button"
        >
          Prev
        </button>
        <button
          className="px-10 py-3 bg-sgreen text-white w-full"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
